Modules/hpe3par_snapshot.py

```diff
@@ -435,13 +468,188 @@ def restore_snapshot_online(
         snapshot_name,
         {})
 
+def create_schedule(
+        client_obj,
+        storage_system_ip,        
+        storage_system_username,
+        storage_system_password,
+        schedule_name,
+        snapshot_name,
+        base_volume_name,
+        read_only,
+        expiration_time,
+        retention_time,
+        expiration_unit,
+        retention_unit,
+        task_freq, task_freq_custom):
+    if storage_system_username is None or storage_system_password is None:
+        return (
+            False,
+            False,
+            "Schedule creation failed. Storage system username or password is \
+null",
+            {})
+    if schedule_name is None:
+        return (
+            False,
+            False,
+            "Schedule create failed. Schedule name is null",
+            {})                 
+    if snapshot_name is None:
+        return (
+            False,
+            False,
+            "Schedule create failed. Snapshot name is null",
+            {})
+    if len(schedule_name) < 1 or len(schedule_name) > 31:
+        return (False, False, "Schedule creation failed. Schedule name must be atleast 1 character and not more than 31 characters", {})
+
+    if len(snapshot_name) < 1 or len(snapshot_name) > 31:
+        return (False, False, "Schedule create failed. Snapshot name must be atleast 1 character and not more than 31 characters", {})
+    if base_volume_name is None:
+        return (
+            False,
+            False,
+            "Schedule create failed. Base volume name is null",
+            {})
+    if len(base_volume_name) < 1 or len(base_volume_name) > 31:
+        return (False, False, "Schedule create failed. Base volume name must be atleast 1 character and not more than 31 characters", {})
+    try:
+        client_obj.login(storage_system_username, storage_system_password)
+        client_obj.setSSHOptions(storage_system_ip, storage_system_username, storage_system_password)
+        expirationHours = convert_to_hours(expiration_time, expiration_unit)
+        retentionHours = convert_to_hours(retention_time, retention_unit)
+
+        freq = "@hourly"
+        if not client_obj.volumeExists(base_volume_name):
+           return (False, False, "Volume does not Exist", {})
+        
+        if retentionHours > expirationHours:
+           return (False, False, "Expiration time must be greater than or equal to retention time", {})   
+       
+        if task_freq_custom:          
+          if ' ' in task_freq_custom:            
+           taskcm = str(task_freq_custom).split()
```

Comment:
what is meant by taskcm? Please separate different words with _ to be consistent with the rest of the code